salt/states/archive.py

```diff
@@ -211,7 +212,7 @@ def extracted(name,
                 ret['result'] = False
                 ret['changes'] = results
                 return ret
-            if __salt__['cmd.retcode']('tar --version | grep bsdtar', python_shell=True) == 0:
+            if re.search('bsdtar', __salt__['cmd.run']('tar --version', python_shell=True)):
```

Comment:
I would do
```python
if 'bsdtar' in  __salt__['cmd.run']('tar --version'):
```
Other than that, this looks great.  Also, @thatch45 may condemn me for picking at code thus.  :-)